### ISSUE TITLE
Recursive include of SDFs

### ISSUE THREAD

**Issue by jbpratt** [RedHatQE/teflo#75]
In a complex scenario, or mono-repo containing numerous scenarios, sharing SDFs is desired. The `include` functionality works great for this, it would be awesome to expand this to handle an included file that also includes others.

```yaml
---
name: Provision child
description: child

provision:
    - name: child
      groups:
        - localhost
      ip_address: 127.0.0.1
      ansible_params:
        ansible_connection: local
---
name: Provision parent
description: includes another SDF

include:
  - provision_child.yml
---
name: Example
description: hello world

include:
  - provision_parent.yml
```

```
❯ teflo validate -s scenario.yml
2021-06-03 11:40:44,786 WARNING Scenario workspace was not set, therefore the workspace is automatically assigned to the current working directory. You may experience problems if files needed by teflo do not exists in the scenario workspace.
2021-06-03 11:40:44,790 INFO

2021-06-03 11:40:44,790 INFO                                TEFLO RUN (START)
2021-06-03 11:40:44,790 INFO -------------------------------------------------------------------------------
2021-06-03 11:40:44,791 INFO  * Data Folder           : .teflo/pqlidmh3j7
2021-06-03 11:40:44,791 INFO  * Workspace             : /home/bpratt/pit/git/teflo_recursive_include
2021-06-03 11:40:44,792 INFO  * Log Level             : info
2021-06-03 11:40:44,792 INFO  * Tasks                 : ['validate']
2021-06-03 11:40:44,793 INFO  * Scenario              : Example
2021-06-03 11:40:44,793 INFO  * Included Scenario(s)  : ['Provision parent']
2021-06-03 11:40:44,793 INFO -------------------------------------------------------------------------------

2021-06-03 11:40:44,794 INFO  * Task    : validate
2021-06-03 11:40:44,794 INFO Sending out any notifications that are registered.
2021-06-03 11:40:44,795 INFO ..................................................
2021-06-03 11:40:44,796 INFO Starting tasks on pipeline: notify
2021-06-03 11:40:44,797 WARNING ... no tasks to be executed ...
2021-06-03 11:40:44,798 INFO ..................................................
2021-06-03 11:40:44,799 INFO Starting tasks on pipeline: validate
2021-06-03 11:40:44,799 INFO --> Blaster v0.4.0 <--
2021-06-03 11:40:44,800 INFO Task Execution: Sequential
2021-06-03 11:40:44,801 INFO Tasks:
2021-06-03 11:40:44,802 INFO 1. Task     : Provision parent
                                Class    : <class 'teflo.tasks.validate.ValidateTask'>
                                Methods  : ['run']
2021-06-03 11:40:44,802 INFO 2. Task     : Example
                                Class    : <class 'teflo.tasks.validate.ValidateTask'>
                                Methods  : ['run']
2021-06-03 11:40:44,803 INFO ** BLASTER BEGIN **
2021-06-03 11:40:44,804 INFO Validating <class 'teflo.resources.scenario.Scenario'> (Provision parent)
2021-06-03 11:40:44,858 INFO Validating <class 'teflo.resources.scenario.Scenario'> (Example)
2021-06-03 11:40:44,908 INFO ** BLASTER COMPLETE **
2021-06-03 11:40:44,908 INFO     -> TOTAL DURATION: 0h:0m:0s
2021-06-03 11:40:44,909 INFO ..................................................
2021-06-03 11:40:44,909 INFO Sending out any notifications that are registered.
2021-06-03 11:40:44,911 INFO ..................................................
2021-06-03 11:40:44,912 INFO Starting tasks on pipeline: notify
2021-06-03 11:40:44,913 WARNING ... no tasks to be executed ...
2021-06-03 11:40:44,921 INFO

2021-06-03 11:40:44,922 INFO                                SCENARIO RUN (END)
2021-06-03 11:40:44,923 INFO -------------------------------------------------------------------------------
2021-06-03 11:40:44,924 INFO  * Duration                       : 0h:0m:0s
2021-06-03 11:40:44,924 INFO  * Passed Tasks                   : ['validate']
2021-06-03 11:40:44,925 INFO  * Results Folder                 : .teflo/.results
2021-06-03 11:40:44,926 INFO  * Included Scenario Definition   : ['.teflo/.results/Provision parent_results.yml']
2021-06-03 11:40:44,926 INFO  * Final Scenario Definition      : .teflo/.results/results.yml
2021-06-03 11:40:44,927 INFO -------------------------------------------------------------------------------
2021-06-03 11:40:44,927 INFO TEFLO RUN (RESULT=PASSED)

pit/git/teflo_recursive_include  v3.9.5(venv) 10s
❯ teflo run -t provision -s .teflo/.results/results.yml
--------------------------------------------------
Teflo Framework v1.2.0
Copyright (C) 2021, Red Hat, Inc.
--------------------------------------------------
2021-06-03 11:40:54,859 WARNING Scenario workspace was not set, therefore the workspace is automatically assigned to the current working directory. You may experience problems if files needed by teflo do not exists in the scenario workspace.
2021-06-03 11:40:54,863 INFO

2021-06-03 11:40:54,863 INFO                                TEFLO RUN (START)
2021-06-03 11:40:54,864 INFO -------------------------------------------------------------------------------
2021-06-03 11:40:54,864 INFO  * Data Folder           : .teflo/ox9ut0cb8e
2021-06-03 11:40:54,864 INFO  * Workspace             : /home/bpratt/pit/git/teflo_recursive_include
2021-06-03 11:40:54,865 INFO  * Log Level             : info
2021-06-03 11:40:54,865 INFO  * Tasks                 : ['provision']
2021-06-03 11:40:54,865 INFO  * Scenario              : Example
2021-06-03 11:40:54,866 INFO  * Included Scenario(s)  : ['Provision parent']
2021-06-03 11:40:54,866 INFO -------------------------------------------------------------------------------

2021-06-03 11:40:54,866 INFO  * Task    : provision
2021-06-03 11:40:54,867 INFO Sending out any notifications that are registered.
2021-06-03 11:40:54,867 INFO ..................................................
2021-06-03 11:40:54,867 INFO Starting tasks on pipeline: notify
2021-06-03 11:40:54,868 WARNING ... no tasks to be executed ...
2021-06-03 11:40:54,869 INFO ..................................................
2021-06-03 11:40:54,869 INFO Starting tasks on pipeline: provision
2021-06-03 11:40:54,869 WARNING ... no tasks to be executed ...
2021-06-03 11:40:54,872 INFO ..................................................
2021-06-03 11:40:54,872 INFO Sending out any notifications that are registered.
2021-06-03 11:40:54,873 INFO ..................................................
2021-06-03 11:40:54,873 INFO Starting tasks on pipeline: notify
2021-06-03 11:40:54,874 WARNING ... no tasks to be executed ...
2021-06-03 11:40:54,878 INFO

2021-06-03 11:40:54,879 INFO                                SCENARIO RUN (END)
2021-06-03 11:40:54,879 INFO -------------------------------------------------------------------------------
2021-06-03 11:40:54,880 INFO  * Duration                       : 0h:0m:0s
2021-06-03 11:40:54,880 INFO  * Passed Tasks                   : ['provision']
2021-06-03 11:40:54,880 INFO  * Results Folder                 : .teflo/.results
2021-06-03 11:40:54,881 INFO  * Included Scenario Definition   : ['.teflo/.results/Provision parent_results.yml']
2021-06-03 11:40:54,881 INFO  * Final Scenario Definition      : .teflo/.results/results.yml
2021-06-03 11:40:54,882 INFO -------------------------------------------------------------------------------
2021-06-03 11:40:54,882 INFO TEFLO RUN (RESULT=PASSED)
```

The scenario passes validation with no error thrown, but the child SDF is still not included.

**Comment by rujutashinde** [comment]
Jira ticket CCITCARBON-319 has been created for this issue. Url : https://projects.engineering.redhat.com/browse/CCITCARBON-319

**Comment by JunqiZhang0** [comment]
closed by #107
